DataOnion/AuthExtensions.cs

```diff
@@ -0,0 +1,96 @@
+using Microsoft.Extensions.DependencyInjection;
+using DataOnion.Auth;
+using StackExchange.Redis;
+
+namespace DataOnion
+{
+    public class RedisConnectionFailedException : RedisException
+    {
+        public RedisConnectionFailedException(
+            string? message = null
+        ) : base(message ?? $"Failed to establish Redis connection.")
+        {
+        }
+    }
```

Comment:
Would it be a good idea to maybe make a separate section for exceptions?

Reply:
i dont know if this exception should be in this codebase

Reply:
@lawhorkl are you suggesting that I don't wrap a RedisException when getting the database? I don't have anything against that, I was just trying to be consistent with our existing codebases.